internal/pkg/apiutil/attribute.go

```diff
@@ -1848,7 +1871,31 @@ func MarshalSRSegments(segs []bgp.TunnelEncapSubTLVInterface) []*any.Any {
 					BFlag: s.Flags&0x10 == 0x10,
 				},
 			}
-			// TODO (sbezverk) Add Type B Segment when SRv6 Binding SID gets finalized.
+		case *bgp.SegmentTypeB:
+			flags := &api.SegmentFlags{
+				VFlag: s.Flags&0x80 == 0x80,
+				AFlag: s.Flags&0x40 == 0x40,
+				SFlag: s.Flags&0x20 == 0x20,
+				BFlag: s.Flags&0x10 == 0x10,
+			}
```

Comment:
Could you change `if else` to just a single `if` 
```
r = &api.SegmentTypeB{
					Flags: flags,
					Sid:   s.SID,
				}
if s.SRv6EBS != nil {
// Assign EndpointBehaviorStructure
}
```

Reply:
done, thanks